[JENKINS-27268] Use agent instead of slave in UI labels, javadoc and logs

amuniz opened the pull request:
[JENKINS-27268](https://issues.jenkins-ci.org/browse/JENKINS-27268)

See [governance meeting](http://meetings.jenkins-ci.org/jenkins/2015/jenkins.2015-04-01-18.02.log.html) for brackground (starting at around 18:29 until 18:55).

`Slave` was literally replaced by `Agent` in most cases. And only two _semantic_ change were done:
1. `Dumb Slave` is now `Permanent Slave`
2. `JNLP slave agent` is now `JNLP Agent`
- [x] Rename UI labels (properties and jelly files), only for english.
- [x] Remove language keys (properties) for other languages (agreed in the PR comments)
- [x] Rename javadocs
- [x] Rename logs and exception messages (which are visible to users or admins)
- [x] Rename "Esclavo" to "Agente" in spanish translation (sorry, I have no knowledge on any other language)

We expect to have others (native speakers) re-adding the removed keys for other languages.

@reviewbybees

amuniz committed:
[JENKINS-27268] Use agent instead of slave in UI labels

varmenise commented:
:bee:

jtnord commented:
I have to :bug: as all the translations (well French!) is still the literal translation of "Slave"

Why don't you just put in properties for American English that use "agent" - and leave the rest of the world to understand that a "slave" != "Human slave"?

jtnord commented on core/src/main/resources/hudson/model/Job/buildTimeTrend_pt_BR.properties:
```diff
@@ -22,7 +22,7 @@
 title={0} Tend\u00eancia de tempo de builds
 Duration=Dura\u00e7\u00e3o
-Slave=SlaveMore\ than\ 1\ builds\ are\ needed\ for\ the\ trend\ report.=\u00c9 necess\u00e1rio mais do que um build para criar o relat\u00f3rio de tend\u00eancia.
+Agent=SlaveMore\ than\ 1\ builds\ are\ needed\ for\ the\ trend\ report.=\u00c9 necess\u00e1rio mais do que um build para criar o relat\u00f3rio de tend\u00eancia.
```
this file is incorrect it should be 2 lines here not one.

amuniz replied:
It was one line before this change, I'm sure that if I would have split this line you would have put a bug for unrelated changes :smiley_cat:

reviewbybees commented:
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

amuniz commented:
> All the translations (well French!) is still the literal translation of "Slave"

I intentionally left translations as they were because I could translate some of them (spanish, for example), but I can not do it for the rest (I don't know if using "agent" in French is correct or not). I will send a specific PR for Spanish once this one is merged. I don't speak Korean, sorry.

> Why don't you just put in properties for American English that use "agent"

It would be confusing. That would lead to american people to talk about "agents" and the rest of the world talking about "slaves". The general change was agreed in the governance meeting that I linked in the description.

jtnord commented:
> It would be confusing. That would lead to american people to talk about "agents" and the rest of the world talking about "slaves"

Which is what we have unless the translations are translated.

There are some translations which literally have "Slave"

jtnord commented:
@amuniz so IMO your options are - fix the translations ([Google](https://translate.google.co.uk/) may help) or remove the translations so they get the English version.

amuniz commented:
@jtnord Removing them is not an option as I see it. I'm going to _suggest_ translations, but I'm sure after that someone from other part of the world will complain about the term used... and this will never end... let's try it.

amuniz commented:
> Which is what we have unless the translations are translated.

Not exacly. We have english by default (currently using _agent_), there are - as average - 8-10 translation files/per jelly, there are roughly 6500 spoken languages in the world today, so only the 0.15% of the languages would still use "slave".

abayer commented:
:bug: on the failing tests - gotta change to agent there too.

But in re: the translations, I disagree with @jtnord - I don't think it's actually a good idea for someone to try translating terms in languages they don't understand in the first place. =) That said, you should probably do the Spanish ones, since you actually do know Spanish. =)

amuniz commented:
@jtnord I have to agree with @abayer. I just started to translate language files and found this line:

```
WindowsSlaveInstaller.ConfirmInstallation=Dette vil installere en Jenkins slave agent som en Windows service, s\u00e5 Jenkins slaven vil starte automatisk n\u00e5r computeren starter op.
```

I can translate the first occurrence as "agent" since it seems to be an English literal, but what about _slaven_? And _노예_ (which is even in unicode in files)?

Fixing the test code.

jtnord commented:
@abayer - but the translations are now wrong - if you can't fix them as you don't have the language skills or trust Google (it does make interesting choices!) then remove the translation so the community can again translate it correctly for you.

If they are not removed then the old translations will still show and then people will still use slave which is the whole point of this.

jtnord commented:
btw - slaven means slave - so you will still have people using the terminology "Slave" 
in this case it is slaven->agenten if Danish (could be Norweigen!) is similar enough to Swedish  :-)

Vlatombe commented:
English -> French :
- slave -> esclave
- agent -> agent

jglick commented:
It does seem safest to simply delete the non-English translations. You can use for example

    cd core && groovy move-l10n.groovy hudson/lifecycle/Messages - 'WindowsSlaveInstaller.ConfirmInstallation'

to avoid lots of manual file edits (and wondering whether your editor is preserving the ISO-8859-1 encoding).

amuniz committed:
[JENKINS-27268] Fix tests after renaming

batmat commented:
+1 with @jglick and @jtnord : IMO better to simply remove all the translations and let translators fix that in the future.

I still see you left some `slave` strings, for example:
* https://github.com/amuniz/jenkins/blob/JENKINS-27268/core/src/main/java/hudson/TcpSlaveAgentListener.java#L69 for example, maybe you did not change those logs on purpose?
* core/src/main/java/hudson/cli/OnlineNodeCommand.java

     @Argument(metaVar="NAME", usage="**Slave** name, or empty string for master"
* core/src/main/java/hudson/console/ConsoleLogFilter.java in javadoc (and in some other places). IMO we agreed to not change the classes currently as it's gonna break just too many things, but even occurrences in code should be replaced as much as possible
* and some other places here and there. I could work on the whole list if needed.

Re-reading the meeting logs, and trying to remember the context as I was present, I'd say basically only renaming files was accepted to be /not/ done.

WDYT?

amuniz commented:
@batmat Yeah, I didn't change anything in Java classes so I missed the log messages. I'll fix that.
And I'm currently removing the translation tags in properties.

Thanks for the feedback!

amuniz committed:
Remove translations using the slave term

amuniz committed:
Spanish translation added

batmat commented:
Great! Thank YOU for the heavy lifting!

amuniz committed:
Slave to Agent renaming - Javadocs and logs

amuniz committed:
Fix bg translation - move-l10n.groovy does not remove multi-line properties

amuniz committed:
Fix tests after renaming in log messages

orrc commented:
@amuniz Can you please edit the PR description to summarise what the wording changes are?

I can see that "slave" has been changed to "agent", but not sure if there's other stuff in the mega diff :)

amuniz commented:
@orrc 

> but not sure if there's other stuff in the mega diff :)

Just "slave" to "agent" changes, only in UI, javadocs and logs, as stated in the PR title. No functional changes, I promise :smile:

amuniz committed:
Merge branch 'master' into JENKINS-27268

Conflicts:
	core/src/main/java/hudson/model/AbstractProject.java

amuniz commented:
There are tests failing because an OOM error... unrelated to this PR IMO. Running fine locally.

amuniz commented:
Trying to get a clean build.

amuniz commented:
@jtnord @abayer please, bee or re-bug please.
@jglick @batmat @orrc looks good to you now?

jtnord commented on core/src/main/resources/hudson/model/Computer/sidepanel_es.properties:
```diff
@@ -22,7 +22,7 @@
 Back\ to\ List=Volver
 Status=Estado
-Delete\ Slave=Borrar esclavo
+Delete\ Slave=Borrar agente
```
This is now "Delete Agent"  - you changed the key in `core/src/main/resources/hudson/model/Computer/sidepanel.jelly`

jtnord commented on core/src/main/resources/hudson/model/Messages_bg.properties:
```diff
@@ -166,38 +162,6 @@ CLI.wait-node-online.shortDescription=\
 CLI.wait-node-offline.shortDescription=\
  \u0418\u0437\u0447\u0430\u043a\u0432\u0430\u043d\u0435 \u043d\u0430 \u043c\u0430\u0448\u0438\u043d\u0430 \u0434\u0430 \u043d\u0435 \u0435 \u043d\u0430 \u043b\u0438\u043d\u0438\u044f.
 
-Computer.Caption=\
```
this looks like a lot - is this no longer used and was just here because it was never deleted before?

amuniz replied:
Possibly, I found others and deleted it (if related to this PR).

jtnord commented on core/src/main/resources/hudson/slaves/ComputerLauncher/main_de.properties:
```diff
@@ -21,6 +21,5 @@
-Launch\ slave\ agent=Slave-Agenten starten
 launchingDescription=Dieser Knoten ist offline, weil Jenkins den dortigen Slave-Agenten nicht starten konnte.
-Relaunch\ slave\ agent=Slave-Agenten neu starten
+Relaunch\ agent=Slave-Agenten neu starten
```
who did the German? this looks like it still says "slave agent..."?

jtnord commented on core/src/main/resources/hudson/slaves/ComputerLauncher/main_da.properties:
```diff
@@ -20,7 +20,6 @@
 launchingDescription=Denne node er ved at starte op
 See\ log\ for\ more\ details=Se loggen for flere detaljer
-Relaunch\ slave\ agent=Genstart slave agent
+Relaunch\ agent=Genstart slave agent
```
has the German been verified?  this looks like it still mentions slaves.

jtnord commented on core/src/main/resources/hudson/slaves/ComputerLauncher/main_ja.properties:
```diff
@@ -21,6 +21,5 @@
 See\ log\ for\ more\ details=\u8A73\u7D30\u306F\u30ED\u30B0\u3092\u53C2\u7167
-Launch\ slave\ agent=\u30B9\u30EC\u30FC\u30D6\u30A8\u30FC\u30B8\u30A7\u30F3\u30C8\u3092\u8D77\u52D5
-Relaunch\ slave\ agent=\u30B9\u30EC\u30FC\u30D6\u30A8\u30FC\u30B8\u30A7\u30F3\u30C8\u3092\u518D\u8D77\u52D5
+Relaunch\ agent=\u30B9\u30EC\u30FC\u30D6\u30A8\u30FC\u30B8\u30A7\u30F3\u30C8\u3092\u518D\u8D77\u52D5
```
Did KK to the review to make sure this doesn't use slave?

jtnord commented on core/src/main/resources/hudson/slaves/JNLPLauncher/main_da.properties:
```diff
@@ -20,10 +20,6 @@
 Connected\ via\ JNLP\ agent.=Forbundet via JNLP agent.
-Or\ if\ the\ slave\ is\ headless\:=Eller hvis slaven er hovedl\u00f8s:
-Run\ from\ slave\ command\ line\:=K\u00f8r fra slavens kommandolinje:
 slaveAgentPort.disabled=TCP port for JNLP slave agent er sl\u00e5et fra.
```
Hmmm

jtnord commented on core/src/main/resources/hudson/slaves/JNLPLauncher/main_da.properties:
```diff
@@ -20,10 +20,6 @@
-Or\ if\ the\ slave\ is\ headless\:=Eller hvis slaven er hovedl\u00f8s:
-Run\ from\ slave\ command\ line\:=K\u00f8r fra slavens kommandolinje:
 slaveAgentPort.disabled=TCP port for JNLP slave agent er sl\u00e5et fra.
-Launch\ agent\ from\ browser\ on\ slave=Start agent fra browser p\u00e5 slaven
```
Hmmm

jtnord commented:
:bug: I don't think the !(Engish|Spanish) translations have all been correctly removed or updated -and it looks like at least one Spanish translation is not correct.

amuniz committed:
Fix spanish translation files

amuniz committed:
Fix remaining occurrences of _slave_

amuniz commented:
@jtnord good catches! Thanks for the review. I've given a second pass and fixed everything I could find.

batmat commented:
LGTM overall. 

Some missed I guess, but a small portion that will progressively be fixed with time.

Still a number of "esclave" strings (french word) in the sources, mostly in the html help files obviously. I think some of those just be removed as a whole like the properties.

To be simply deleted IMO:
* core/src/main/resources/hudson/model/Node/help-labelString* (non en)
* core/src/main/resources/hudson/model/Slave/help-remoteFS* non EN
* certainly most of core/src/main/resources/hudson/node_monitors/**/*.html

Some remarks, maybe you left those out on purpose? :

* core/src/main/java/hudson/model/Computer.java:1340:            w.println("Master to slave");
* core/src/main/java/hudson/model/Computer.java:1344:            w.println("\n\n\nSlave to master");
* core/src/main/java/hudson/model/Computer.java:1404
* core/src/main/java/hudson/node_monitors/package.html:26:Code that monitors the health of slaves
* core/src/main/java/jenkins/model/Jenkins.java:4266:     * Automatically try to launch an agent when Jenkins is initialized or a new slave is created.
* core/src/main/java/jenkins/security/s2m/package-info.java:2
* core/src/main/resources/hudson/model/ComputerSet/new.jelly:26:  "New slave" page
* core/src/main/resources/hudson/model/Node/help-labelString.html many occurrences
* core/src/main/resources/hudson/model/Slave/help-remoteFS.html idem
IMO, we should also add an explanation in the `Slave` javadoc for future maintainers, and so on, about the history of that naming, and why slave has been left at some places, and so on. (The choice having been done: only non-classes get replaced, etc.)
* core/src/main/resources/hudson/model/UsageStatistics/help-usageStatisticsCollected.html:11
* core/src/main/resources/hudson/security/GlobalSecurityConfiguration/config_nl.properties:27 ...and some other files there
* core/src/main/resources/hudson/slaves/CommandLauncher/help-command.html

I can provide a full list if need be. But basically I found them all using `git grep -i slave` or `git grep -n -i slave`.

Thanks!

amuniz committed:
Replace _slave_ occurrences in HTML files

amuniz committed:
Add history information about this renaming in Slave class javadocs

amuniz committed:
Merge branch 'JENKINS-27268' of github.com:amuniz/jenkins into JENKINS-27268

amuniz commented:
@batmat I used Eclipse "Search in Files", but filtered by `*.properties,*.jelly,*.groovy`, so I missed `*.html`, thanks for pointing out.

For HTML I don't like the idea of removing all that translation work (it's not like removing a single key from a properties file but much more text that we would not want to lose), so I replaced "Slave" by "Agent" blindly without language specific semantics, so that anyone - native - can fix it in the future without having to re-write the full HTML page. WDYT?

> core/src/main/java/hudson/model/Computer.java:1340: w.println("Master to slave");
> core/src/main/java/hudson/model/Computer.java:1344: w.println("\n\n\nSlave to master");

I intentionally kept this occurrences, since they are used to generate a dump of the export table and probably it is used by 3rd parties in automated processes.

> IMO, we should also add an explanation in the Slave javadoc for future maintainers

+1, fixed.

batmat commented:
> For HTML I don't like the idea of removing all that translation work (it's not like removing a single key from a properties file but much more text that we would not want to lose), so I replaced "Slave" by "Agent" blindly without language specific semantics, so that anyone - native - can fix it in the future without having to re-write the full HTML page. WDYT?

Not sure. For example, doing so misses the existing french translations. So only for french, it leaves 23 files with the `esclave` word which is likely to stay for a few months unchanged (years?).

```
$ git grep -l -i esclave
core/report-l10n.rb
core/src/main/resources/hudson/model/AbstractItem/help-slaveAffinity_fr.html
core/src/main/resources/hudson/model/LoadStatistics/main_fr.properties
core/src/main/resources/hudson/model/Messages_fr.properties
core/src/main/resources/hudson/model/Node/help-labelString_fr.html
core/src/main/resources/hudson/model/Node/help-name_fr.html
core/src/main/resources/hudson/model/Node/help-numExecutors_fr.html
core/src/main/resources/hudson/model/Slave/help-remoteFS_fr.html
core/src/main/resources/hudson/model/UsageStatistics/help-usageStatisticsCollected_fr.html
core/src/main/resources/hudson/security/GlobalSecurityConfiguration/help-slaveAgentPort_fr.html
core/src/main/resources/hudson/slaves/CommandLauncher/help-command_fr.html
core/src/main/resources/hudson/slaves/JNLPLauncher/help-vmargs_fr.html
core/src/main/resources/hudson/slaves/JNLPLauncher/main_fr.properties
war/src/main/webapp/help/project-config/custom-workspace_fr.html
war/src/main/webapp/help/system-config/master-slave/availability_fr.html
war/src/main/webapp/help/system-config/master-slave/clock_fr.html
war/src/main/webapp/help/system-config/master-slave/demand/idleDelay_fr.html
war/src/main/webapp/help/system-config/master-slave/demand/inDemandDelay_fr.html
war/src/main/webapp/help/system-config/master-slave/description_fr.html
war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel_fr.html
war/src/main/webapp/help/system-config/master-slave/jnlpSecurity_fr.html
war/src/main/webapp/help/system-config/master-slave/numExecutors_fr.html
war/src/main/webapp/help/system-config/master-slave/usage_fr.html
```

batmat commented:
Anyway, thinking about it again, I'm not sure there's a simple solution anyway. Let's get it out this way, and we'll review translations in a second pass, ping the translation team, tweet and so on to try and look for left out [translated] `slave` occurrences.

:+1: 

![](http://cdn.meme.am/instances/500x/66726617.jpg)

amuniz commented:
@batmat agree. Thanks for helping!
@jtnord @abayer what about your bugs there? Do you keep them?

abayer commented:
Tests are passing now, so :bee: =)

kohsuke commented:
I echo @orrc 's comment that the summary of changes would be useful.

For example, I see that the new term is "permanent agent", which has never mentioned in the original IRC log nor in the discussion of this pull request. FWIW, what was originally proposed was "static build agent".

Presumably some people have gotten together and decided what the new terms should be (and I hope that included Patrick), and if that's the case I don't intend to bike shed on the name, but it would be helpful to have the name change summarized for translators.

batmat commented:
Hi @kohsuke, crawled the issue and didn't find (nor remember) any Patrick.
Would you care to clarify which Patrick it is? Thanks!
Le 2 févr. 2016 10:16 PM, "Kohsuke Kawaguchi" <notifications@github.com> a
écrit :

> I echo @orrc <https://github.com/orrc> 's comment that the summary of
> changes would be useful.
>
> For example, I see that the new term is "permanent agent", which has never
> mentioned in the original IRC log nor in the discussion of this pull
> request. FWIW, what was originally proposed was "static build agent".
>
> Presumably some people have gotten together and decided what the new terms
> should be (and I hope that included Patrick), and if that's the case I
> don't intend to bike shed on the name, but it would be helpful to have the
> name change summarized for translators.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/jenkins/pull/2007#issuecomment-178824815>.
>

jtnord commented:
> Sorry, we could not display the entire diff because too many files (391) changed.

So I need to review this locally - which I doubt I will have full time for - but I am still concerned that we are not removing things that have translations of slave that will live forever.
Perhaps these files should be renamed .lang.slave_rename so that people can find them in source code if they are developers to perform a easier update - but where we have no developers we are still prompting for a crowd source using the inbluilt provide a translation link.

batmat commented:
> Perhaps these files should be renamed .lang.slave_rename so that people can find them in source code 

@jtnord renaming the .html files might indeed be a good compromise, good idea. 

> For example, I see that the new term is "permanent agent", which has never mentioned in the original IRC log nor in the discussion of this pull request. FWIW, what was originally proposed was "static build agent".

@kohsuke just looked at that, and if didn't miss some, there's indeed (only) one `permanent` occurrence in `core/src/main/resources/hudson/slaves/DumbSlave/newInstanceDetail.properties`. 
Basically IIUC _dumb_ was replaced by _permanent_ in that file, which was indeed not agreed on in that meeting or later. So fixing that by using 'static' instead shouldn't be hard.

_(Side note: I know there's no general agreement about that, but I hope the commits of this PR will get squashed somehow before merging :-).)_

daniel-beck commented:
Why is this targeting the master branch?

amuniz commented:
@daniel-beck 

> Why is this targeting the master branch?

Yeah, I'll close this PR a refile it against 2.0, but let's finish the review here as I don't want to lose the conversation.

@kohsuke 

> I see that the new term is "permanent agent", which has never mentioned in the original IRC 

Proposed by @olivergondza 
```
18:35:12 <ogondza> I suggest "static slave" or "permanent slave"
```
and @jglick 
```
18:38:51 <jglick> “Dumb” was never a useful adjective in this context, though. “Static” or “permanent” or “manual” etc. would be more helpful.
```
and I like it too. Anyway, if _static_ sounds better for you all, I'm ok to change it - as @batmat pointed it's only one occurrence.

> the summary of changes would be useful.

Done.

jtnord commented:
>> @daniel-beck
>> Why is this targeting the master branch?

>Yeah, I'll close this PR a refile it against 2.0, but let's finish the review here as I don't want to lose the conversation.

Why should this target the 2.0 branch?  It is not breaking any backwards compatibility - and for a 2.0 branch I would have expected more, like deprecating all the classes in the slave package and introducing intermediaries in the hierarchy in a new package.

rtyler commented:
@jtnord This ticket is part of the [2.0 terminology sweep](https://issues.jenkins-ci.org/browse/JENKINS-31095).

Changing the names of things in a point release, while totally compatible, is a fairly big change for the user experience.

amuniz commented:
Ok. Let's recap, and please, give your +1 or -1 on each:

 1. @kohsuke suggests _Static Build Agent_ instead of _Permanent Agent_
 2. @jtnord suggests to rename HTML translation files (the ones containing the term _slave_) to `*.lang.slave_rename` instead of keeping them with a possible wrong translation.

abayer commented:
I'm not a big fan of "Permanent Agent" - it's weird. "Static Build Agent" would be better. 

The translation thing...wouldn't that break all other existing translations? I'm confused.

amuniz commented:
@abayer 

> The translation thing...wouldn't that break all other existing translations? I'm confused.

No, @jtnord just suggests to rename `XXX_Y.html` to `XXX_Y.html.lang.slave_rename` only for HTML files containing the _slave_ term, so we don't remove them (as it was the intention initially) and they are not _visible_ to Jenkins.

batmat commented:
@amuniz 
1. Static build agent sounds better/clearer to me than "Permanent", but I'm no native speaker so my opinion is half-important or less
2. +1 with @jtnord 's renaming proposal

amuniz committed:
[JENKINS-27268] Remove HTML files containing _slave_

amuniz commented:
@batmat @jtnord renaming of HTML files done. I you are ok with current status I'll file a new PR targeting the 2.0 branch (as @daniel-beck requested).

daniel-beck commented:
> so we don't remove them (as it was the intention initially) and they are not visible

I don't understand why these files are kept around. This looks like a strategy from before using an SCM, because old versions can be looked up any time even when they're deleted.

oleg-nenashev commented:
I don't see any value in the new "agent" term. It causes much work in plugins. Why don't we just make the "node" or "build node" term the official one. It maps well to the current class structure, so it should work better

batmat commented:
@oleg-nenashev seeing this PR at first, I thought the same. And re-reading the IRC log clarified it: node was somehow rejected because `master` is also a node and so on.

orrc commented:
@oleg-nenashev @batmat Renaming to *anything* other than "slave" will require work in numerous plugins, but I think "agent" makes a lot of sense.

An "agent" is someone that has the power to perform some actions on your behalf.
The word "node" doesn't carry as much meaning for me; it's a point in a graph, or just some computer in a network.

I think avoiding "*build* node" or "*build* agent", and just calling them "agents" also helps to remove the idea that Jenkins is only for *building* software — agents could be *testing* or *deploying* or doing any number of other automated tasks.

<sub>(Sorry for the PR spam)</sub>

slide commented:
@orrc +1

amuniz commented:
Hey guys, let's get this done. There was already an agreement on using _Agent_, please do not discuss that again.

For the last commit removing the HTML files, I have been thinking about it and yeah, we have a CVS for it... so we don't need to keep that files which could be potentially there forever. I will remove them and then will squash all commits into one so it's easy to refer to removed files in the future (in order to recover them if necessary).

oleg-nenashev commented:
Yep, I accept the decision. Makes sense to perform the renaming in bundled plugins at least, but I'm OK with merging.

batmat commented:
Ok. But as for squashing commits, IMO you could have a dedicated commit for
the .html removals. But anyways.
Le 11 févr. 2016 5:31 PM, "Oleg Nenashev" <notifications@github.com> a
écrit :

> Yep, I accept the decision. Makes sense to perform the renaming in bundled
> plugins at least, but I'm OK with merging.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/jenkins/pull/2007#issuecomment-182945080>.
>

amuniz commented:
@batmat sure, I'll do it when refiling this against 2.0

KostyaSha commented:
> Why should this target the 2.0 branch? It is not breaking any backwards compatibility - and for a 2.0 branch I would have expected more, like deprecating all the classes in the slave package and introducing intermediaries in the hierarchy in a new package.

Such changes killing users documentations and books about jenkins. 

Instead of actualising translations you are adding terminology mess. :-1: 

> I think avoiding "build node" or "build agent", and just calling them "agents" also helps to remove the idea that Jenkins is only for building software — agents could be testing or deploying or doing any number of other automated tasks.

Master<->slave is default architecture pattern for any sw. 

PS How much :beer: you got at FOSDEM?

batmat commented:
@KostyaSha not sure what you are against. 
If this is against renaming per-se, actually, this PR is not about discussing if rename should happen. 

As stated in the associated issue, as decided in the related gov meeting (link provided by @amuniz in the PP description), this change *will* happen. 
We are only discussing details and "implementation" here.

KostyaSha commented:
Meetings are only for those who living in convenient TZ for meetings, or has no family and etc. Good projects do discussions and document RFCs in offline mode before, so everybody may participate. I found PR, i commented it.

stephenc commented:
@kostyasha this is a bit OT but I suspect it gets to the heart of some of the complaints I see from you.

The Jenkins project currently does not have a visible hierarchy of decision making and conflict resolution. 

By this I mean that, currently the only way to resolve contentious choices is to have a board vote.

If we take another project I am involved in (Apache maven) there can be perceived similar issues... Though that project at least has the tradition of the `[VOTE]` threads.

In the Jenkins project, we don't even have a tradition of `[VOTE]` threads, so secondary concerns such as who's eligible to vote, does anyone have more of a vote than others, etc do not even stet to arise. (Aside: in maven, everyone has a vote, committers can decide to ignore non-committer votes, the PMC has binding votes on matters of releasing source code and ensuring that committers don't excessively overrule a user majority without just cause)

One model that @kosuke an the board seem to be following is delegating areas of responsibility to specify persons for a 1 year duration. That person is (in my terminology) effectively the decision "tzar" for that area of responsibility. A better analogy comes from Ancient Rome where they would occasionally make specific persons in charge of all decisions for a fixed time period... But sadly I forget the name used for that term (Julius Caesar was elected one before he was in the triumvirate if somebody wants to look up the terminology) now if you abuse such trust delegation, the board may revoke (in case of repeated major abuse) or just not renew (for minor abuse). The person is supposed to use the trust delegation to help build the community while actually making decisions where decisions need to be made.

So for example, @daniel-beck is currently in charge of all things security. @oleg-nenashev is in charge of remoting, etc. (I hope there is a page somewhere that lists all the current assignments, but I have not checked)

So that is my understanding of the model the board is following...

As such, there is a person responsible for the 2.0 effort. They decided the name change was above their interpretation of the delegates powers, so they asked the board to confirm the choice (none of this may be how it was worded, but that is really what the intent should have been - even if stumble upon by accident)

My view is that we need to get the community into a habit of at least seeing the consultation and decision making processes... So, eg if I were responsible for a specific area I would probably have `[RFC]` threads to gather opinions, and then some sort of consensus probe thread similar to a vote (but not called vote) with a final decision post to declare the outcome.

Others can do as they want within their sphere of control

HTH

kohsuke commented:
This change has been proposed & discussed as a part of the Jenkins 2.0 proposal, and after considering various feedbacks/implications, we made a conscious decision to proceed with this change and I still stand by that.

I haven't seen anything in this thread that's new & significant to reconsider the decision. I haven't read every comment carefully, but it appears to me that Antonio has done everything he has been asked for, and if not, such minor follow-up changes can be done in separate PRs.

daniel-beck commented:
@kohsuke Please address the comments that said this should be refiled against 2.0 once accepted. [1](https://github.com/jenkinsci/jenkins/pull/2007#issuecomment-179124481) [2](https://github.com/jenkinsci/jenkins/pull/2007#issuecomment-179158767) [3](https://github.com/jenkinsci/jenkins/pull/2007#issuecomment-179168660). I see absolutely no reason to merge this into master.

orrc commented:
This has been [merged](https://github.com/jenkinsci/jenkins/commit/f40f9bbbad7a6c2e795f854a475689c12638e560) into the [2.0 branch](https://github.com/jenkinsci/jenkins/tree/2.0).

kohsuke commented:
Expanding on what orrc wrote above for book-keeping sake.

It was originally incorrectly merged into the master branch, which I eventually nullified via `git push -f` to avoid diamond cross merge problem later when the master branch is re-merged into the 2.0 branch (that is, such a merge will cause the change made in this PR to be entirely reverted.)

All is good now, but there was a mis-step for a brief period of time.

KostyaSha commented:
@stephenc i think you are the best understand and commenting issues that usually happens with miss-communications and miss-organisations in this project :+1: I'm trying follow dev maillist, but as it usually spammed with different things. Maybe i missed again something.

I firstly hear that somebody is already elected and leading some fields (that of course good steps). And jira issues are getting 2.0-rejected/new task without any clear visibility (still have no idea whether real painful issues would be resolved as part of 2.0). https://issues.jenkins-ci.org/secure/BrowseProjects.jspa?selectedCategory=all for security lead is @kohsuke (really?)

Remoting has no lead https://issues.jenkins-ci.org/browse/JENKINS/component/15489/
The most voted issues in remoting not solved and not even started
https://issues.jenkins-ci.org/browse/JENKINS-23271?focusedCommentId=246932&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-246932 

How this assigned leads are driving their fields? CB interests firstly? Whom should people(?)/me(?) ask/report questions about such activities? Is there any task planning for this fields? Who is participating in task planning?

As person that maintained and wrote different internal docs about jenkins with screenshots and supported end users, i don't understand how it was possible to merge or +1 such changes to master! ( @daniel-beck :+1: as usual for QA).

stephenc commented:
@KostyaSha this is not really the right place for this discussion, but in any case @oleg-nenashev was [announced as the remoting lead](https://groups.google.com/forum/#!searchin/jenkinsci-dev/oleg$20remoting/jenkinsci-dev/gndP582STPM/u0erzsvLCQAJ)

The other leads I am aware of are:

* @daniel-beck for security
* @olivergondza for the LTS line
* somebody was selected for the new website, but I cannot recall exactly who

The jenkins community board appoints the leads, and people who are assigned such a lead are supposed to fulfill that role **for the benefit of the community**. From some of the internal debates I have seen, most engineers in CloudBees are all together too good at arguing on behalf of the community, so while there is a risk of corporate bias, so far I do not believe the risk to have been realized.

KostyaSha commented:
@stephenc thanks. I began to remember something about it.
